GitUI/UserControls/RevisionGrid/Columns/AvatarColumnProvider.cs

```diff
@@ -46,55 +48,70 @@ public override void OnCellPainting(DataGridViewCellPaintingEventArgs e, GitRevi
 
             Column.Width = e.CellBounds.Height;
 
-            var padding = DpiUtil.Scale(2);
-            var imageSize = e.CellBounds.Height - padding - padding;
+            var imageSize = e.CellBounds.Height - _padding - _padding;
 
             Image? image;
             var imageTask = _avatarProvider.GetAvatarAsync(revision.AuthorEmail, revision.Author, imageSize);
 
-            if (imageTask.Status == TaskStatus.RanToCompletion)
+            Rectangle rect = new(
+                e.CellBounds.Left + _padding,
+                e.CellBounds.Top + _padding,
+                imageSize,
+                imageSize);
+
+            bool isLoaded = imageTask.Status == TaskStatus.RanToCompletion;
+            if (isLoaded)
             {
 #pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
                 image = imageTask.Result;
 #pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
             }
             else
             {
+                // First time, draw resized and cache placholder image
+                if (_buffer == null)
+                {
+                    _buffer = new Bitmap(imageSize, imageSize);
+                    Graphics g = Graphics.FromImage(_buffer);
+                    g.InterpolationMode = InterpolationMode.HighQualityBicubic;
+                    g.DrawImage(Images.User80, 0, 0, imageSize, imageSize);
+                }
+
                 // Image is not yet loaded -- use a placeholder for now
-                image = Images.User80;
+                e.Graphics.DrawImageUnscaled(_buffer, rect);
 
                 // One the image has loaded, invalidate for repaint
                 imageTask.ContinueWith(
                     t =>
                     {
                         if (t.Status == TaskStatus.RanToCompletion)
                         {
-                            _revisionGridView.Invalidate();
+                            _revisionGridView.Invalidate(rect);
                         }
                     }, TaskScheduler.Current)
                     .FileAndForget();
+                return;
             }
 
-            Rectangle rect = new(
-                e.CellBounds.Left + padding,
-                e.CellBounds.Top + padding,
-                imageSize,
-                imageSize);
-
-            var container = e.Graphics.BeginContainer();
-            e.Graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
```

Comment:
This line should not be removed I guess.

Reply:
I can reinstate it but for me this line is useless because the image is retrieved from the avatar service already add the right size and so don't need any resize...

Reply:
Then it is OK for me. I was not sure whether it got lost when extracting the code for the placeholder.